test/module/irohad/synchronizer/synchronizer_test.cpp

```diff
@@ -210,103 +196,96 @@ TEST_F(SynchronizerTest, ValidWhenValidChain) {
+  gate_outcome.get_subscriber().on_next(VoteOther{commit_message});
 
   ASSERT_TRUE(wrapper.validate());
 }
 
 /**
- * @given A valid block that cannot be applied directly
- * @when process_commit is called
- * @then observable of retrieveBlocks must be evaluated four times:
- *   - to validate whole chain
- *   - to validate last block of chain (x2)
- *   - to create a vector
+ * @given A commit from consensus and initialized components
+ * @when gate have voted for other block
+ * @then retrieveBlocks called again after unsuccessful download attempt
  */
 TEST_F(SynchronizerTest, ExactlyThreeRetrievals) {
   auto commit_message = makeCommit();
+  rxcpp::observable<std::shared_ptr<shared_model::interface::Block>>
+      commit_message_blocks = rxcpp::observable<>::just(commit_message);
 
   DefaultValue<expected::Result<std::unique_ptr<MutableStorage>, std::string>>::
       SetFactory(&createMockMutableStorage);
   EXPECT_CALL(*mutable_factory, createMutableStorage()).Times(1);
   EXPECT_CALL(*mutable_factory, commit_(_)).Times(1);
-  EXPECT_CALL(*consensus_gate, on_commit())
-      .WillOnce(Return(rxcpp::observable<>::empty<network::Commit>()));
-  EXPECT_CALL(*chain_validator, validateChain(_, _))
-      .WillOnce(Return(false))
-      .WillOnce(testing::Invoke([](auto chain, auto &) {
-        // emulate chain check
-        chain.as_blocking().subscribe([](auto) {});
-        return true;
-      }));
+  EXPECT_CALL(*consensus_gate, onOutcome())
+      .WillOnce(Return(gate_outcome.get_observable()));
+  EXPECT_CALL(*chain_validator, validateChain(_, _)).WillOnce(Return(true));
   EXPECT_CALL(*block_loader, retrieveBlocks(_))
-      .WillOnce(Return(rxcpp::observable<>::create<std::shared_ptr<
-                           shared_model::interface::Block>>([commit_message](
-                                                                auto s) {
-        static int times = 0;
-        if (times++ > 4) {
-          FAIL() << "Observable of retrieveBlocks must be evaluated four times";
-        }
-        s.on_next(commit_message);
-        s.on_completed();
-      })));
+      .WillOnce(Return(rxcpp::observable<>::empty<
+                       std::shared_ptr<shared_model::interface::Block>>()))
+      .WillOnce(Return(commit_message_blocks));
 
   init();
 
   auto wrapper =
       make_test_subscriber<CallExact>(synchronizer->on_commit_chain(), 1);
   wrapper.subscribe();
 
-  synchronizer->process_commit(commit_message);
+  gate_outcome.get_subscriber().on_next(VoteOther{commit_message});
 
   ASSERT_TRUE(wrapper.validate());
 }
 
 /**
- * @given commit from the consensus and initialized components
- * @when synchronizer fails to download block from some peer
- * @then it will try until success
+ * @given initialized components
+ * @when gate have got reject on block
+ * @then synchronizer output is also reject with related block height
  */
-TEST_F(SynchronizerTest, RetrieveBlockTwoFailures) {
```

Comment:
Why is this test case removed? It looks valid, however it seems better to check that it will be called n+1 times if required, where n is number of signatures in commit message.